code/game/machinery/computer/dropship_weapons.dm

```diff
@@ -196,497 +123,660 @@
+		to_chat(weapon_operator, SPAN_WARNING("Vehicle has to be idle to allow Fire Mission editing and creation."))
+		return FALSE
+
+	for(var/datum/cas_fire_mission/mission in firemission_envelope.missions)
+		if(firemission_name == mission.name)
+			to_chat(weapon_operator, SPAN_WARNING("Fire Mission name must be unique."))
+			return FALSE
+	//everything seems to be fine now
+	firemission_envelope.generate_mission(firemission_name, firemission_length)
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_delete_firemission(mob/weapon_operator, firemission_tag)
+	if(!skillcheck(weapon_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapon_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+	if(firemission_tag > firemission_envelope.missions.len)
+		to_chat(weapon_operator, SPAN_WARNING("Fire Mission ID corrupted or already deleted."))
+		return FALSE
+	if(selected_firemission == firemission_envelope.missions[firemission_tag])
+		to_chat(weapon_operator, SPAN_WARNING("Can't delete selected Fire Mission."))
+		return FALSE
+	var/result = firemission_envelope.delete_firemission(firemission_tag)
+	if(result != 1)
+		to_chat(weapon_operator, SPAN_WARNING("Unable to delete Fire Mission while in combat."))
+		return FALSE
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_select_firemission(mob/weapon_operator, firemission_tag)
+	if(!skillcheck(weapon_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapon_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+	if(firemission_envelope.stat > FIRE_MISSION_STATE_IN_TRANSIT && firemission_envelope.stat < FIRE_MISSION_STATE_COOLDOWN)
+		to_chat(weapon_operator, SPAN_WARNING("Fire Mission already underway."))
+		return FALSE
+	if(firemission_tag > firemission_envelope.missions.len)
+		to_chat(weapon_operator, SPAN_WARNING("Fire Mission ID corrupted or deleted."))
+		return FALSE
+	if(selected_firemission == firemission_envelope.missions[firemission_tag])
+		selected_firemission = null
+	else
+		selected_firemission = firemission_envelope.missions[firemission_tag]
+	return TRUE
+
+/obj/structure/machinery/computer/dropship_weapons/proc/ui_firemission_change_offset(mob/weapons_operator, fm_tag, weapon_id, offset_id, offset_value)
+	if(!skillcheck(weapons_operator, SKILL_PILOT, SKILL_PILOT_TRAINED)) //only pilots can fire dropship weapons.
+		to_chat(weapons_operator, SPAN_WARNING("A screen with graphics and walls of physics and engineering values open, you immediately force it closed."))
+		return FALSE
+
+	var/result = firemission_envelope.update_mission(fm_tag, weapon_id, offset_id, offset_value)
+	if(result != null && result != 0)
```

Comment:
A "Good" result is actually currently 1; and it should be unnecessary to null check this.

Since it confused you, it really ought to instead use the defines `FIRE_MISSION_ALL_GOOD` `FIRE_MISSION_CODE_ERROR` and `FIRE_MISSION_NOT_EXECUTABLE` but note that this will mean `/obj/structure/machinery/computer/dropship_weapons/proc/update_mission(mission_id, weapon_id, offset_step, offset)` also needs to be updated.

Reply:
Addressed this. Ran a quick test and I'm still able to run firemissions.